### PR TITLE
[stable/external-dns]: added service monitor

### PR DESCRIPTION
**Pull request by caarlos0**
#### What this PR does / why we need it:

Adds an option to create a servicemonitor when installing the chart.

Also fixes readme default for `metrics.podAnnotations`

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/external-dns/README.md [hunk 1]
```diff
@@ -149,7 +149,8 @@ The following table lists the configurable parameters of the external-dns chart
...
 | `readinessProbe`                   | Deployment Readiness Probe                                                                               | See `values.yaml`                                        |
 | `metrics.enabled`                  | Enable prometheus to access external-dns metrics endpoint                                                | `false`                                                  |
-| `metrics.podAnnotations`           | Annotations for enabling prometheus to access the metrics endpoint                                       | {`prometheus.io/scrape: "true",prometheus.io/port: "7979"`} |
+| `metrics.serviceMonitor.enabled`   | Enable prometheus-operator service monitor                                                               | `false`                                                  |
+| `metrics.podAnnotations`           | Annotations for enabling prometheus to access the metrics endpoint                                       | `{}`                                                     |
```
Comment (miguelaeh) [inline stable/external-dns/README.md]: This parameter does not exist in the `values.yaml` nor `values-production.yaml` could you add it, please?

File: stable/external-dns/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  labels: {{ include "external-dns.labels" . | nindent 4 }}
+spec:
+  endpoints:
+    - port: http
+      path: /metrics
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels: {{ include "external-dns.labels" . | nindent 4 }}
```
Comment (miguelaeh) [inline stable/external-dns/templates/servicemonitor.yaml]: This indentation should be 6 spaces not 4.

File: stable/external-dns/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.metrics.serviceMonitor.enabled }}
```
Comment (miguelaeh) [inline stable/external-dns/templates/servicemonitor.yaml]: Change to:
```diff
- {{- if .Values.metrics.serviceMonitor.enabled }}
+ {{- if and (.Values.metrics.enabled) (.Values.metrics.serviceMonitor.enabled) }}
```

File: stable/external-dns/templates/servicemonitor.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "external-dns.fullname" . }}
+  labels: {{ include "external-dns.labels" . | nindent 4 }}
```
Comment (miguelaeh) [inline stable/external-dns/templates/servicemonitor.yaml]: Add:
```
  {{- if .Values.metrics.serviceMonitor.namespace }}
  namespace: {{ .Values.metrics.serviceMonitor.namespace }}
  {{- end }}
```
And the fields to the `values.yml`

Comment (miguelaeh) [inline stable/external-dns/templates/servicemonitor.yaml]: Add to labels:
```
    {{- range $key, $value := .Values.metrics.serviceMonitor.selector }}
    {{ $key }}: {{ $value | quote }}
    {{- end }}
```
And the corresponding field to the `values.yaml`

File: stable/external-dns/templates/servicemonitor.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  endpoints:
+    - port: http
```
Comment (miguelaeh) [inline stable/external-dns/templates/servicemonitor.yaml:15]: Add:
```
    {{- if .Values.metrics.serviceMonitor.interval }}
    interval: {{ .Values.metrics.serviceMonitor.interval }}
    {{- end }}
```
And the fields to the `values.yaml`

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you very much for the PR!!!!!!!
Could you please take a look to my comments?

**Comment by mikhailadvani** [comment]
really need this. I can pick it up from here if needed

**Comment by caarlos0** [comment]
> eally need this. I can pick it up from here if needed

please, go ahead... I forgot about this PR, sorry :/

**Comment by caarlos0** [comment]
(if you can, please ping me on the new PR so I take a look as well :) )

**Comment by zanhsieh** [comment]
@caarlos0 
Would you guys resolve conflicting files please? Thank you.
stable/external-dns/Chart.yaml
stable/external-dns/README.md

File: stable/external-dns/templates/servicemonitor.yaml (hunk 4 above)
Comment (caarlos0) [reply stable/external-dns/templates/servicemonitor.yaml]: I don't see why, the selector will always be the labels used on the service anyway 🤔

File: stable/external-dns/README.md (hunk 1 above)
Comment (caarlos0) [reply stable/external-dns/README.md]: done.,

**Comment by caarlos0** [comment]
fixed the issues

**Comment by miguelaeh** [comment]
Hi @caarlos0 ,
Can you please fix the conflicts?

**Pull request closed without merging**